lib/bindings/python/rust/lib.rs

```diff
@@ -162,6 +163,7 @@ fn register_llm<'p>(
     let model_name = model_name.map(|n| n.to_string());
     let router_mode = router_mode.unwrap_or(RouterMode::RoundRobin);
     let router_config = RouterConfig::new(router_mode.into(), KvRouterConfig::default());
+    let custom_template_path_owned = custom_template_path.map(|s| s.to_string());
```

Comment:
Could you also do the PathBuf conversion here (you can make a pathbuf directly from a `&str`), and then check if it `.exists()`? Then we can error early.